modules/quaketerm/term.go

```diff
@@ -0,0 +1,164 @@
+package launcher
+
+import (
+	"time"
+
+	"fyshos.com/fynedesk"
+	"fyshos.com/fynedesk/internal/ui"
+	wmTheme "fyshos.com/fynedesk/theme"
+	"github.com/fyne-io/terminal"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/canvas"
+	"fyne.io/fyne/v2/container"
+	"fyne.io/fyne/v2/driver/desktop"
+	"fyne.io/fyne/v2/theme"
+)
+
+const (
+	delay     = time.Second / 25
+	termTitle = "Quake Terminal " + ui.SkipTaskbarHint
+	height    = 240
+	step      = 40
+)
+
+var termMeta = fynedesk.ModuleMetadata{
+	Name:        "\"Quake\" (hover) terminal",
+	NewInstance: newTerm,
+}
+
+type term struct {
+	shown bool
+	win   fynedesk.Window
+	ui    fyne.Window
+}
+
+func (t *term) Destroy() {
+}
+
+func (t *term) Metadata() fynedesk.ModuleMetadata {
+	return termMeta
+}
+
+func (t *term) Shortcuts() map[*fynedesk.Shortcut]func() {
+	return map[*fynedesk.Shortcut]func(){
+		&fynedesk.Shortcut{Name: "Open Quake Terminal", KeyName: fyne.KeyBackTick, Modifier: fynedesk.UserModifier}: func() {
+			t.toggle()
+		}}
+}
+
+func (t *term) createTerm() {
+	win := fyne.CurrentApp().Driver().(desktop.Driver).CreateSplashWindow()
+	win.SetTitle(termTitle)
+
+	bg := canvas.NewRectangle(theme.BackgroundColor())
+	img := canvas.NewImageFromResource(theme.NewDisabledResource(theme.ComputerIcon()))
+	img.FillMode = canvas.ImageFillContain
+	img.SetMinSize(fyne.NewSize(200, 200))
+	over := canvas.NewRectangle(wmTheme.WidgetPanelBackground())
+	matchTheme(bg, over)
+
+	console := terminal.New()
+	win.SetContent(container.NewStack(bg, img, over, console))
+	win.Canvas().Focus(console)
+	t.ui = win
+
+	go func() {
+		err := console.RunLocalShell()
+		if err != nil {
+			fyne.LogError("Failed to open terminal", err)
+		}
+		t.hide() // terminal exited
+
+		t.createTerm() // reset for next usage
```

Comment:
I have tmux auto-launch from my .bashrc (with a check to make sure it isn't already running.) The way this is written here will cause my tmux sessions to pile up rather quickly, if I'm understanding it correctly.

Looks like this behavior might not play nicely with pinentry-ncurses for Yubikey users. Is there a way to only spawn the shell JIT, right before the user actually sees it, rather than leaving it open in the background 24/7 if the module is enabled?